pkg/util/azureclient/resources.go

```diff
@@ -11,6 +11,7 @@ import (
 type DeploymentsClient interface {
 	CreateOrUpdate(ctx context.Context, resourceGroupName string, deploymentName string, parameters resources.Deployment) (result resources.DeploymentsCreateOrUpdateFuture, err error)
 	Client
+	DeploymentClient() resources.DeploymentsClient
```

Comment:
@mjudeikis @asalkeld this one looks like a limitation of our sdk interfaces.

Reply:
it is. But this way we can mock any interfaces with mock data for unti tests